rfc/0009-qsl-structure.md

```diff
@@ -0,0 +1,129 @@
+
+Additionally, it defines that mods will depend on QSL as a fat jar, and Loom is expected to detect what (sub/super)modules are being used and automatically mark them to be downloaded at runtime.
+## Motivation
+### Prior Art
+Organization of code into visible, transparent modules is a zero-sum game involving QSL maintainers, mod developers, and the end users.  
+Here is a quick comparison between current systems that exist today:
+- Fabric API
+	- 
+	- Large amount of single-purpose, small, and sometimes interconnected modules, consumed by developers and users as a manually downloaded fat jar.
+	- Maintainers
+		-
+		- Benefits
+			- Replacing large swaths of code is easy
+			- The small parts keep scope in check, avoiding "dumping grounds"
+		- Drawbacks
+			-  Everybody just uses the fat jar
+			-  API visibility is low since modules don't usually inter-connect
+				-  Compare Forge's Item.setModelRenderer (devoldified) to using a registry for everything
+			-  The massive amount of modules makes navigating FAPI confusing
+			-  Even though there are small modules, modules are often very tiny--Extensions of Items might be split across Content Registries, Item API, Tool Item API, 
+- Forge
+	-
+	- Benefits
+		- Monolithic API makes adding new features straightforward
+	- Drawbacks
+		- It is more difficult to remove parts of code as Mojang breaks them
+		- Lack of compartmentalization leads to seemingly unrelated APIs being intertwined
+		- No api/impl split makes it difficult to know what part of the code you are actually supposed to interact with
+### Goals
+- Be mobile and quick to update like Fabric
+- Have high API visibility like Forge
+- Find a balance between Forge's confusingly monolithic and Fabric's confusingly modular structures
+- Have teams in charge of different parts of QSL's code
+- Avoid making users download and run code they don't need
+## Explanation
+This is an overview of the top-level structure of QSL. Libraries are included, with the names of the FAPI modules that would be under its scope. The actual structure of the modules in each library is up to the team, and it is not a goal to define how they must structure their modules.
+### Libraries
+Each library would be its own folder in the QSL monorepo, with a dedicated team responsible for triage, review, and accepting/denying pull requests. In contrast to Fabric or Forge's system of only one person signing off for the entire standard library, no authority higher than the people on the team would be required to sign off on the PR before it is merged. More clarification on team structure is expected to come in a future RFC.
+
+Libraries only exist for internal structure--there is no fat jar for, say, all `worldgen` modules, neither is there a fat jar for all of QSL. This also means that QSL and its libraries have no versioning.
+### Modules
+Modules follow the following general rules:
+- Maven group of `org.quiltmc.qsl.$library`
```

Comment:
I have to say that might end up a bit confusing for anyone having to deal with a maven group.